sdk/eventhub/event-hubs/src/eventHubProducerClient.ts

```diff
@@ -261,8 +274,13 @@ export class EventHubProducerClient {
    * @throws Error if the underlying connection has been closed, create a new EventHubProducerClient.
    * @throws AbortError if the operation is cancelled via the abortSignal.
    */
-  getEventHubProperties(options: GetEventHubPropertiesOptions = {}): Promise<EventHubProperties> {
-    return this._client.getProperties(options);
+  async getEventHubProperties(
```

Comment:
This doesn't need to be an `async` function since all it does is return a Promise, does it? The one benefit I see for making this `async` is if we think `managementSession` might be `undefined` so error handling is consistent for the caller.
Same question applies to `getPartitionProperties`.

Reply:
I made `getPartitionIds` async coz I fell into the trap of "replace all .then with async/await". Then made the others async just to be consistent.

Would you have had a knee jerk reaction to make `getPartitionIds` async if you saw this:

```js
  getPartitionIds(options: GetPartitionIdsOptions = {}): Promise<Array<string>> {
    return this._context.managementSession!.getEventHubProperties({
      ...options,
      retryOptions: this._clientOptions.retryOptions
    }).then(eventHubProperties => {
      return eventHubProperties.partitionIds;
    });
  }
```

Reply:
I don't think I'd have the same reaction there since we're not doing anything with the value in the `then` callback except returning something.

But the only real reason I prefer not using `async` in these scenarios is because I find stepping through the TS pollyfilled code to be somewhat difficult, and otherwise think of this as sort of a micro optimization. I won't push on this if we want to favor consistency via `async` here.

Reply:
I agree 100% that this was a micros optimization. Reverted the async/await change

Reply:
Fixed in https://github.com/Azure/azure-sdk-for-js/pull/9520/commits/26b61431f2b2e7bbeb5c5fb3acf185a7dbf188a8